File: src/iPad/MyVoteLV/MyVoteLV/PollOption.m
//
//  PollOption.m
//  MyVoteLV
//
//  Created by Kasey Schindler on 1/16/14.
//  Copyright (c) 2014 Magenic. All rights reserved.
//

#import "PollOption.h"

@implementation PollOption

- (instancetype)initPollOptionWithDictionary:(NSDictionary *)dictionary
{
    self = [super init];
    
    if (self) {
        // create poll option
        NSNumber *holder = [dictionary objectForKey:@"PollOptionID"];
        self.pollOptionID = [CommonMethods validValueForJSONNumber:holder];
        
        // option text
        self.optionText = [dictionary objectForKey:@"OptionText"];
        
        // option position
        holder = [dictionary objectForKey:@"OptionPosition"];
        self.optionPosition = [CommonMethods validValueForJSONNumber:holder];
        
        // poll response id
        holder = [dictionary objectForKey:@"PollResponseID"];
        self.pollResponseID = [CommonMethods validValueForJSONNumber:holder];
        
        // is option selected
        holder = [dictionary objectForKey:@"IsOptionSelected"];
        self.isOptionSelected = holder.boolValue;
    }
    
    return self;
}

@end

File: testAutomation/TestAutomation/Models/WebService/PollOption.cs
namespace Models.WebService
{
	public sealed class PollOption
	{

        public PollOption(int? v1, int? v2, short? v3, string v4, bool v5)
        {
            this.PollOptionID = v1;
            this.PollID = v2;
            this.OptionPosition = v3;
            this.OptionText = v4;
            this.Selected = v5;
        }

        public int? PollOptionID { get; set; }
		public int? PollID { get; set; }
		public short? OptionPosition { get; set; }
		public string OptionText { get; set; }
        public bool Selected { get; set; }
    }
}
